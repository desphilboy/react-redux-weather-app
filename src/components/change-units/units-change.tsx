import React from "react";
import { connect } from "react-redux";
import { setUnits } from "../../store/actions";
import { UnitChangerDiv } from "./change-units.style";

export const View = ({ changeUnits, units }) => (
	<UnitChangerDiv>
		<label>Unit types:</label>
		<select name="units" onChange={changeUnits} value={units}>
			<option value="metric">Metric</option>
			<option value="imperial">Imperial</option>
		</select>
	</UnitChangerDiv>
);

const mapDispatch = (dispatch) => ({
	changeUnits: (event) => dispatch(setUnits(event.target.value)),
});

const mapState = (state) => ({
	units: state.units,
});

export default connect(mapState, mapDispatch)(View);
