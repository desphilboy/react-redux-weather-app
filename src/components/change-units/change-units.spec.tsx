import React from "react";
import { View } from "./units-change";
import renderer from "react-test-renderer";

describe("Units Changing Component", () => {
	describe("View only", () => {
		const changeUnitMock = jest.fn();

		it("Renders correctly", () => {
			const wrapper = renderer.create(
				<View changeUnits={changeUnitMock} />
			);
			expect(wrapper.toJSON()).toMatchSnapshot();
		});
	});
});
