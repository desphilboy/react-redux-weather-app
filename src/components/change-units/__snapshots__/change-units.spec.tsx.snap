// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Units Changing Component View only Renders correctly 1`] = `
<div
  className="sc-bdnylx fHbBEQ"
>
  <label>
    Unit types:
  </label>
  <select
    name="units"
    onChange={[MockFunction]}
  >
    <option
      value="metric"
    >
      Metric
    </option>
    <option
      value="imperial"
    >
      Imperial
    </option>
  </select>
</div>
`;
